File: brute_force/2531.c
#include <stdio.h>
#include <string.h>

#define MAX(x, y) (x) > (y) ? (x) : (y)

int solution(int N, int d, int k, int c, int *sushis)
{
    int max_sushi = 0;
    int i, j;
    int check[3001] = {0, };
    int eat_sushi;
    int sushi;

    for (i = 0; i < N; i++) {
        memset(check, 0, d + 1);
        check[c] = 1;
        eat_sushi = 1;

        for (j = i; j < i + k; j++) {
            sushi = sushis[j % N];

            if (!check[sushi]) {
                eat_sushi++;
            }
            check[sushi]++;
        }
        max_sushi = MAX(max_sushi, eat_sushi);
    }

    return max_sushi;
}

int main(void)
{
    int N, d, k, c;
    int sushis[30000] = {0, };
    int i;

    scanf("%d %d %d %d", &N, &d, &k, &c);
    for (i = 0; i < N; i++) {
        scanf("%d", &sushis[i]);
    }

    printf("%d\n", solution(N, d, k, c, sushis));
}

File: brute_force/2531.py
''' [회전초밥]
'''

import sys
from collections import defaultdict

def solution(N: list, d: int, k: int, c: int, sushis: list) -> int:
    max_sushi = 0
    for i in range(N):
        # 쿠폰의 초밥을 먼저 먹이고 시작
        check = [0] * (d + 1)
        check[c] = 1 
        eat_sushi = 1
        # 초밥을 연속해서 먹는 경우의 수 계산
        for j in range(i, i + k):
            sushi = sushis[j % N]

            if not check[sushi]:
                eat_sushi += 1
            check[sushi] += 1
        max_sushi = max(max_sushi, eat_sushi)
    
    return max_sushi    

# N: 접시의 수
# d: 초밥의 가짓수
# k: 연속해서 먹는 접시의 수
# c: 쿠폰 번호
N, d, k, c = map(int, sys.stdin.readline().split())
sushis = [int(sys.stdin.readline()) for _ in range(N)]
sys.stdout.write(str(solution(N, d, k, c, sushis)) + '\n')